game/scripts/vscripts/items/farming/greater_power_treads.lua

```diff
@@ -105,28 +105,30 @@ function modifier_item_greater_power_treads:OnCreated( event )
   self.stat = spell:GetSpecialValueFor( "bonus_stat" )
   self.bonus_damage = spell:GetSpecialValueFor( "bonus_damage" )
   self.all_stats = spell:GetSpecialValueFor( "all_stats" )
+
+  if IsServer() then
+    self.creepDamageMelee = spell:GetSpecialValueFor( "creep_damage_melee" )
+    self.creepDamageRanged = spell:GetSpecialValueFor( "creep_damage_ranged" )
+    self.creepDamageMeleeIllusion = spell:GetSpecialValueFor( "creep_damage_melee_illusion" )
+    self.creepDamageRangedIllusion = spell:GetSpecialValueFor( "creep_damage_ranged_illusion" )
+    -- hopefully we don't do a thing where damage type is
+    -- defined by current stat
+    self.damageType = spell:GetAbilityDamageType()
+
+    -- this is probably a bit weird
+    -- but it's a method that came to me and it's also kinda cool
+    -- so!
+    self.damageTable = { 
+      { self.creepDamageMelee, self.creepDamageMeleeIllusion, },
+      { self.creepDamageRanged, self.creepDamageRangedIllusion, },
+    }
+  end
 end
 
 --------------------------------------------------------------------------------
 
 function modifier_item_greater_power_treads:OnRefresh( event )
```

Comment:
Afaik you can write  `modifier_item_greater_power_treads.OnRefresh =  modifier_item_greater_power_treads.OnCreated` instead of this. Example here: https://github.com/OpenAngelArena/oaa/blob/master/game/scripts/vscripts/abilities/oaa_glaives_of_wisdom.lua#L56

Reply:
I'm aware. I just prefer this method since it makes it simpler to add OnCreated or OnRefresh-specific behavior afterwards should either become necessary.